inst/include/common/fims_math.hpp

```diff
@@ -220,6 +268,420 @@ inline const Type ad_max(const Type &a, const Type &b, Type C = 1e-5) {
   return (a + b + fims_math::ad_fabs(a - b, C)) * static_cast<Type>(.5);
 }
 
+    /**
+     * \ingroup NormalDistribution
+     * 
+     * @brief The probability density function for a standard normal distribution.
+     *
+     * \f$
+     * f(x) = \frac{1}{\sqrt{2\pi\theta^2}}exp(-\frac{(x-\mu)^2}{2\theta^2})
+     * \f$<br>
+     * <b>Where,</b><br>
+     *  \f$\ \mu \f$  is the mean.<br>
+     *  \f$\ \theta\f$ is the standard deviation.<br>
+     * 
+     *  @image html http://upload.wikimedia.org/wikipedia/commons/7/74/Normal_Distribution_PDF.svg
+     * 
+     *  <br> <b>Source:</b> http://en.wikipedia.org/wiki/Normal_distribution<br>
+     * 
+     * 
+     * 
+     * @author Matthew Supernaw
+     * @date June 6, 2013
+     *  
+     * @param x
+     * @param mean
+     * @param sd
+     * @return relative likelihood for this random variable to have value x.
+     */
+    template<class T>
+    const T dnorm(const T &x, const T &mean, const T &sd, bool ret_log = false) {
+      
+        // Check if the standard deviation is valid
+        if (sd <= static_cast<T> (0)) {
+            throw std::invalid_argument("Standard deviation must be positive.");
+        }
+
+        // Constants
+        const T inv_sqrt_2pi = 1.0 / std::sqrt(2.0 * M_PI);
+
+        // Compute the exponent part
+        T z = (x - mean) / sd;
+        T exponent = -1.0*static_cast<T> (0.5) * z * z;
+
+        // Compute log density
+        if (ret_log) {
+            return exponent - log(sd) - log(inv_sqrt_2pi);
```

Comment:
It should be + log(inv_sqrt_2pi) since you are taking the inverse inside the log. Also, it's recommended to write in logspace then take the exp() if not log to simplify calculations, eg.:
ret = -1/2*ln(2pi) - ln(sd) - exponent
if(ret_log) ret else exp(ret)

Reply:
I believe this has been corrected in e06e0363f0262ae6f90588d5a1d97440cd60087c, @Andrea-Havron-NOAA can you confirm and resolve or comment.